### PR TITLE
Trying different workflows

### PR DESCRIPTION
**Pull request by Hartorn**

### CODE REVIEW

File: .github/workflows/build_backend.yml [hunk 1]
```diff
@@ -0,0 +1,232 @@
...
+name: Full CI
+on:
+  push:
+    branches:
+      - main
+  pull_request:
+  workflow_dispatch:
+    inputs:
+      run-integ:
+        description: 'If integ test should be run'
```
Comment (andreybavt) [inline .github/workflows/build_backend.yml:13]: Was "integration test" label too long for a GH UI?

File: .github/workflows/build_backend.yml [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+jobs:
+  pre-check:
+    name: Pre check
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - name: Extract branch name
```
Comment (andreybavt) [inline .github/workflows/build_backend.yml:38]: you don't use it anymore, do you?

File: .github/workflows/build_backend.yml (hunk 1 above)
Comment (Hartorn) [reply .github/workflows/build_backend.yml:13]: No, changing it

File: .github/workflows/build_backend.yml (hunk 2 above)
Comment (Hartorn) [reply .github/workflows/build_backend.yml]: I was, but I can managed without it

**Comment by andreybavt** [comment]
I think if tests pass and `ingeg` is renamed to `integration` in various places it should be fine to merge

**Comment by Hartorn** [comment]
> I think if tests pass and `integ` is renamed to `integration` in various places it should be fine to merge

Was sure I did it, I must have not pushed it before.

**Comment by Hartorn** [comment]
@andreybavt Should be all good now

**Pull request merged**